keps/sig-network/20190603-EndpointSlice-API.md

```diff
@@ -127,7 +128,15 @@ type AddressType string
 
 const (
     // AddressTypeIP represents an IP Address.
+    // This address type has been deprecated and has been replaced by the IPv4
+    // and IPv6 adddress types. New resources with this address type will be
+    // considered invalid. This will be fully removed in 1.18.
+    // +deprecated
     AddressTypeIP = AddressType("IP")
+    // AddressTypeIPv4 represents an IPv4 Address.
+    AddressTypeIPv4 = AddressType("IPv4")
```

Comment:
It would be great if we can limit what describes IPFamily `IPv4` and `IPv6` to what is already in types.

Reply:
Interesting point.  \We have two overlapping sets (I guess it's a subset).  We don't want to imply that "FQDN" is a valid value for service IPFamily.  I kind of like types being documentation for this, so perhaps:

```
AddressTypeIPv4 = AddressType(corev1.IPv4Protocol)
AddressTypeIPv6 = AddressType(corev1.IPv6Protocol)
AddressTypeFQDN = AddressType("FQDN")
```

I don't If we REALLY need to, we could extract a common pkg for API constants and const define all of these strings, so at least all representations stay consistent?